restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/impl/OrExpressionDocumentsResolver.java

```diff
@@ -0,0 +1,189 @@
+
+  private final Set<FilterExpression> children;
+
+  private final List<AbstractSearchQueryBuilder> queryBuilders;
+
+  private final ExecutionContext context;
+
+  public OrExpressionDocumentsResolver(Or<FilterExpression> expression, ExecutionContext context) {
+    this.expression = expression;
+    this.children = getChildren(expression);
+    this.queryBuilders = buildQueries(this.children);
+    this.context = createContext(context, expression);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Flowable<RawDocument> getDocuments(
+      QueryExecutor queryExecutor,
+      DocsApiConfiguration configuration,
+      String keyspace,
+      String collection,
+      Paginator paginator) {
+    DataStore dataStore = queryExecutor.getDataStore();
+
+    // find the max size of columns in all queries and use that for now
+    String[] columns =
+        queryBuilders.stream()
+            .map(qb -> columnsForQuery(qb, configuration.getMaxDepth()))
+            .max(Comparator.comparingInt(o -> o.length))
+            .orElseGet(() -> QueryConstants.ALL_COLUMNS_NAMES.apply(configuration.getMaxDepth()));
+
+    return Flowable.fromIterable(queryBuilders)
+        .flatMap(
+            queryBuilder ->
+                RxUtils.singleFromFuture(
+                        () -> {
+                          // TODO column per query if possible
+                          // String[] columns = columnsForQuery(queryBuilder,
+                          // configuration.getMaxDepth());
+                          BuiltQuery<? extends BoundQuery> query =
+                              queryBuilder.buildQuery(
+                                  dataStore::queryBuilder, keyspace, collection, columns);
+                          return dataStore.prepare(query);
+                        })
+                    .toFlowable())
+        .toList()
+        .cache()
+        .flatMapPublisher(
+            preparedQueries -> {
+              // find all with empty bind values
+              List<BoundQuery> boundQueries =
+                  preparedQueries.stream().map(Query::bind).collect(Collectors.toList());
+
+              // execute them all by respecting the paging state
+              return queryExecutor.queryDocs(
+                  boundQueries,
+                  configuration.getSearchPageSize(),
+                  paginator.getCurrentDbPageState(),
+                  context);
+            })
+        .filter(
+            doc -> {
+              // now we can get doc as a result of the persistence or in memory query
+              // if we can locate that it was a result of the persistence ones, then nothing is
+              // needed anymore
+              // we can do this by finding any row that does not have any path related column
+              // TODO enable if solved in the query executor
+              // boolean noPaths = doc.rows().stream().anyMatch(r ->
+              // !r.columnExists(QueryConstants.P_COLUMN_NAME.apply(0)));
```

Comment:
As discussed, I think we can determine whether a doc was produced by a qurry with a "persistence" predicate by validating whether it has any of the `*_value` columns.

Alternatively, and this may actually be cleaner, we could update `QueryExecutor` to link `RawDocument` objects to  queries that provided rows for the doc. In that case, if any of the queries with "persistence" predicates contributed to the doc, we do not have to evaluate rules in memory.

I suppose we could introduce a boolean flag on `RawDocument`, e.g. `matched`, then add a function parameter to `QueryExecutor` for mapping `Query` object to the `matched` flag, and have `QueryExecutor` invoke that function on every row collected (inside `PagingStateTracker.track()` probably) and aggregate the final value for the `matched` flag. WDYT?

By comparison, using rows with different selected column sets is risky, because rows differing only in `*_value` columns are considered "equal" by the merge logic and there's no telling which of those equal rows will be stored in the doc object.

Reply:
On a second though, we can probably do this optimization later.

For now, if query with "persistence" predicates return `*_value` columns anyway, re-evaluating their expression in memory is a bit of extra work, but it should not change the filter outcome, right? So, it might be ok, to merge as is for now, and optimize in a follow-up PR.

OR conditions are a new feature, so having a bit of extra perf. overhead initially is probably ok. WDYT?